### PR TITLE
Check auto-generated conversions in pre-commit hook

### PR DESCRIPTION
**Pull request by wojtek-t**
Also followup from #8443 

cc @thockin

### CODE REVIEW

File: hack/verify-generated-conversions.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+fi
+
+APIROOT="${KUBE_ROOT}/pkg/api"
+TMP_APIROOT="${KUBE_ROOT}/_tmp/pkg/api"
+_tmp="${KUBE_ROOT}/_tmp"
+
+mkdir -p "_tmp/pkg"
+cp -a "${APIROOT}" "${TMP_APIROOT}"
+
+"${KUBE_ROOT}/hack/update-generated-conversions.sh"
```
Comment (thockin) [inline hack/verify-generated-conversions.sh]: won't this update in-place and leave that as outstanding changes in my git workspace?

Comment (wojtek-t) [inline hack/verify-generated-conversions.sh]: That was true. I changed this script so that it reverts to the initial contents.

**Comment by wojtek-t** [comment]
PTAL

File: hooks/pre-commit [hunk 2]
```diff
@@ -99,4 +99,16 @@ else
 fi
 echo "${reset}"
 
+echo -ne "Checking for conversions that need updating... "
+if ! hack/verify-generated-conversions.sh > /dev/null; then
+  echo "${red}ERROR!"
+  echo "Some conversions are out of sync between CLI and markdown."
```
Comment (thockin) [inline hooks/pre-commit]: Error message is from the docs check :)

File: hack/verify-generated-conversions.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+_tmp="${KUBE_ROOT}/_tmp"
+
+mkdir -p "${_tmp}"
+cp -a "${APIROOT}" "${TMP_APIROOT}"
+
+"${KUBE_ROOT}/hack/update-generated-conversions.sh"
+echo "diffing ${APIROOT} against freshly generated conversions"
+ret=0
+diff -Naupr -I 'Auto generated by' "${APIROOT}" "${TMP_APIROOT}" || ret=$?
+cp -a ${TMP_APIROOT} "${KUBE_ROOT}/pkg"
```
Comment (thockin) [inline hack/verify-generated-conversions.sh]: This feels brittle but passable.

File: hooks/pre-commit (hunk 2 above)
Comment (wojtek-t) [inline hooks/pre-commit]: Done.

File: hack/verify-generated-conversions.sh (hunk 3 above)
Comment (wojtek-t) [inline hack/verify-generated-conversions.sh]: I agree it's hacky, but I hope it's fine for now.

**Comment by wojtek-t** [comment]
PTAL

**Comment by thockin** [comment]
LGTM, not merging due to freeze

File: hack/verify-generated-conversions.sh [hunk 4]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Copyright 2014 The Kubernetes Authors All rights reserved.
```
Comment (nikhiljindal) [inline hack/verify-generated-conversions.sh]: 2014 -> 2015

Comment (wojtek-t) [inline hack/verify-generated-conversions.sh]: Done.

**Pull request merged**